File: 2020/Python/day6.py
from aoc import *

def solve(inp):
    inp = inp.split("\n\n")
    part1 = part2 = 0
    for group in inp:
        ans = [set(x) for x in group.split()]
        part1 += len(set.union(*ans))
        part2 += len(set.intersection(*ans))
    return part1, part2


run(solve)

File: 2020/Nim/day6.nim
import aoc
import strutils, sequtils

func toSet(ans: string): set[char] =
    for c in ans:
        result.incl(c)

func solve*(input:string): (int, int) =
    let data = input.split("\n\n")
    var part1, part2: int
    for group in data:
        var union, intersection: set[char]
        for person in group.splitLines:
            let personAns = person.toSet
            union = union + personAns
            if intersection.card == 0:
                intersection = personAns
            else:
                intersection = intersection * personAns
        part1 += union.card
        part2 += intersection.card
    return (part1, part2)
                